src/Logjam/Message.php

```diff
@@ -527,30 +416,108 @@ public function getDbCalls()
 
     /**
      * Adds a log line
-     * @param int $severity
-     * @param string $timestamp
-     * @param string $info
+     * @param LineInterface $line
+     * 
      * @return $this
      */
-    public function addLine($severity, $timestamp, $info)
+    public function addLine(LineInterface $line)
     {
-        $this->lines[] = [$severity, $timestamp, $info];
-
-        if ($this->getSeverity() == self::SEVERITY_UNKOWN) {
-            $this->setSeverity($severity);
-        } else {
-            $this->setSeverity(max($severity, $this->getSeverity()));
-        }
-
+        $this->lines[] = $line;
+        
         return $this;
     }
 
     /**
      * Returns the log lines
-     * @return array
+     * 
+     * @return LineInterface[]
      */
     public function getLines()
     {
         return $this->lines;
     }
+
+    /**
+     * @return bool
+     */
+    public function hasLines()
+    {
+        return count($this->lines) > 0;
+    }
+
+    /**
+     * @param $requestInformation
+     * 
+     * @return $this
+     */
+    public function setRequestInformation($requestInformation)
+    {
+        $this->requestInformation = $requestInformation;
+        
+        return $this;
+    }
+    
+    
+    /**
+     * @return RequestInformationInterface
+     */
+    public function getHttpRequestInformation()
+    {
+        return $this->requestInformation;
+    }
+
+    /**
+     * Specify data which should be serialized to JSON
+     * @link http://php.net/manual/en/jsonserializable.jsonserialize.php
+     * @return mixed data which can be serialized by <b>json_encode</b>,
+     * which is a value of any type other than a resource.
+     * @since 5.4.0
```

Comment:
:) Please don't copy the parent header here. Since should refer to a package version.